File: @LV1/1.5/cups/cups.go
package main

import (
	"fmt"
	"sort"
	"strconv"
	"unicode"
)

type cup struct {
	radius int
	name   string
}

type cups []cup

func (c cups) Len() int {
	return len(c)
}

func (c cups) Swap(i, j int) {
	c[i], c[j] = c[j], c[i]
}

func (c cups) Less(i, j int) bool {
	return c[i].radius < c[j].radius
}

func main() {
	var n int
	var a, b string
	var l cups

	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%s %s", &a, &b)
		if unicode.IsDigit(rune(a[0])) {
			tmp, _ := strconv.Atoi(a)
			l = append(l, cup{radius: tmp, name: b})
		} else {
			tmp, _ := strconv.Atoi(b)
			l = append(l, cup{radius: 2 * tmp, name: a})
		}
	}

	sort.Sort(l)
	for _, c := range l {
		fmt.Println(c.name)
	}
}

File: @LV1/1.5/cups/cups.cpp
#include <algorithm>
#include <cctype>
#include <cstdio>
#include <string>
#include <vector>
using namespace std;

typedef pair<int, string> is;

int main() {
    int n, tmp;
    char a[1000], b[1000];
    vector<is> l;

    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        scanf("%s %s", a, b);
        if (isdigit(a[0])) {
            l.push_back(is(atoi(a), b));
        } else {
            l.push_back(is(2 * atoi(b), a));
        }
    }

    sort(l.begin(), l.end());
    for (auto c : l) {
        printf("%s\n", c.second.c_str());
    }
}
